sdk/core/core-lro/test/testClient.test.ts

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import assert from "assert";
+import { delay, SimpleTokenCredential, WebResource, HttpHeaders } from "@azure/core-http";
+import { TestClient } from "./utils/testClient";
+import { PollerStoppedError } from "../src";
+
+const testHttpHeaders: HttpHeaders = new HttpHeaders();
+const testHttpRequest: WebResource = new WebResource();
+const basicResponseStructure = {
+  headers: testHttpHeaders,
+  parsedBody: {},
+  request: testHttpRequest,
+  status: 200
+};
+const initialResponse = {
+  ...basicResponseStructure,
+  parsedBody: {
+    started: true
+  }
+};
+const doFinalResponse = {
+  ...basicResponseStructure,
+  parsedBody: {
+    doFinalResponse: true
+  }
+};
+const finalResponse = {
+  ...basicResponseStructure,
+  parsedBody: {
+    finished: true
+  }
+};
+
+describe("Long Running Operations - custom client", function() {
+  it("can automatically poll a long running operation with one promise", async function() {
+    const client = new TestClient(new SimpleTokenCredential("my-test-token"));
+    client.setResponses([initialResponse, doFinalResponse, finalResponse]);
+
+    const poller = await client.startLRO();
+
+    // synchronously checking the operation state
+    await poller.poll();
+    assert.ok(poller.operation.state.started);
+
+    // Checking the serialized version of the operation
+    let serializedOperation = JSON.parse(poller.toJSON());
+    assert.ok(serializedOperation.state.started);
+
+    // Waiting until the operation completes
+    const result = await poller.done();
+    const { properties, state } = poller.operation;
+
+    assert.ok(properties.initialResponse!.parsedBody.started);
+    assert.ok(properties.previousResponse!.parsedBody.finished);
+    assert.ok(state.completed);
+    assert.equal(result, "Done");
+  });
+
+  it("can automatically poll a long running operation with more than one promise", async function() {
```

Comment:
the test description talks about auto polling but it looks like this test body is a manual polling test?

Reply:
nah, this is automatic polling, but with more than one call to poll(). Since poll() always return the promise of the current poll loop, we can wait for the next poll several times by calling to poll() repeatedly.